internal/utils/telegraf/emitter/baseemitter/baseemitter.go

```diff
@@ -0,0 +1,224 @@
+// sends them through Output
+type BaseEmitter struct {
+	lock   *sync.Mutex
+	Output types.Output
+	Logger *log.Entry
+	// omittedTags are tags that should be removed from measurements before
+	// being processed
+	omittedTags map[string]bool
+	// addTags are tags that should be added to all measurements
+	addTags map[string]string
+	// Telegraf has some junk events so we exclude all events by default
+	// and can enable them as needed by using IncludeEvent(string) or
+	// IncludeEvents([]string).
+	// You should look up included metrics using Included(string)bool.
+	included map[string]bool
+	// excluded metrics and events that should not be emitted.
+	// You can add metrics and events to exclude by name using
+	// ExcludeDatum(string) and ExcludeData(string).  You should look up
+	// excluded events and metrics using Excluded(string)bool
+	excluded map[string]bool
+}
+
+// AddTag adds a key/value pair to all measurement tags.  If a key conflicts
+// the key value pair in AddTag will override the original key on the
+// measurement
+func (B *BaseEmitter) AddTag(key string, val string) {
+	B.lock.Lock()
+	B.addTags[key] = val
+	B.lock.Unlock()
+}
+
+// AddTags adds a map of key value pairs to all measurement tags.  If a key
+// conflicts the key value pair in AddTags will override the original key on
+// the measurement.
+func (B *BaseEmitter) AddTags(tags map[string]string) {
+	for k, v := range tags {
+		B.AddTag(k, v)
+	}
+}
+
+// IncludeEvent a thread safe function for registering an event name to include
+// during emission. We disable all events by default because Telegraf has some
+// junk events.
+func (B *BaseEmitter) IncludeEvent(name string) {
+	B.lock.Lock()
+	B.included[name] = true
+	B.lock.Unlock()
+}
+
+// IncludeEvents a thread safe function for registering a list of event names to
+// include during emission. We disable all events by default because Telegraf
+// has some junk events.
+func (B *BaseEmitter) IncludeEvents(names []string) {
+	for _, name := range names {
+		B.IncludeEvent(name)
+	}
+}
+
+// Included - A thread safe function for checking if events should be included
+// during emission.  We disable all events by default because Telegraf has some
+// junk events.
+func (B *BaseEmitter) Included(name string) (included bool) {
+	B.lock.Lock()
+	included = B.included[name]
+	B.lock.Unlock()
+	return
+}
+
+// ExcludeDatum adds a name to the list of metrics and events to
+// exclude
+func (B *BaseEmitter) ExcludeDatum(name string) {
+	B.lock.Lock()
+	B.excluded[name] = true
+	B.lock.Unlock()
+}
+
+// ExcludeData adds a list of names the list of metrics and events
+// to exclude
+func (B *BaseEmitter) ExcludeData(names []string) {
+	for _, name := range names {
+		B.ExcludeDatum(name)
+	}
+}
+
+// Excluded - A thread safe function for checking if events or metrics should be
+// excluded from emission
+func (B *BaseEmitter) Excluded(name string) (excluded bool) {
```

Comment:
`IsExcluded` would be clearer.  Also should it be part of the `Emitter` interface?

Reply:
Will rename.  It could be added to the interface, but I don't think it's something that monitors/clients will directly invoke.  I think really the only code that might invoke it outside of baseemitter is any extension of baseemitter.  In that case it is an exported Function so any packages that extend it should have access to do what they need with it behind the interface.